main.go

```diff
@@ -127,6 +130,15 @@ func main() {
 	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
 	exitOnError(err, "unable to create Kubernetes client")
 
+	apiextensionsClient, err := clientset.NewForConfig(kubeConfig)
+	if err != nil {
+		exitOnError(err, "unable to create apiextensionsClient")
+	}
+	crdName := "rayclusters.ray.io"
+	if err := checkCRDAvailability(apiextensionsClient, crdName); err != nil {
```

Comment:
How is it different from the call to `hasAPIResourceForGVK` that's done a bit after.

Reply:
TBH I only noticed that new function just now after rebasing. - The main difference is that with the Kubernetes API Extensions client we can directly check for the required CRD, which is more focused on the scope of this issue.

With that said, running the CFO from main branch with the `hasAPIResourceForGVK` approach, for some reason, without the rayclusters CRD installed it doesn't fail the initialisation of the CFO pod.

Reply:
Functionally these two methods does very much the same. It doesn't fail on main only because we want it, we just do not start the RayCluster controller at the moment

`hasAPIResoureForGVK` has the advantage that it doesn't require permission to read CRDs.

Reply:
That's true, a good advantage. One question in case I'm mistaken, do we really want to start the CFO anyways without the RC Controller? Even after applying the CRD, the CFO won't attempt to start the controller. We would remain in the same scenario where the user would need to restart the pod. Moreover, I couldn't find any error messages displayed in the CFO logs for it.

Reply:
The idea was that there may be other controllers orthogonal to KubeRay, like in #491. But it's true at the moment we could exit in that case.

Reply:
I see, thanks for clarifying on that, I wasn't sure. I refactored the logic to exit and not initialise the pod unless the CRD is present. Let me know what you think. I know this is a temporary change too until the RC Controller is moved to KubeRay.

Reply:
Current behaviour now: if the RayCluster CRD is not available, the CFO will fail to start and attempt to restart until the CRD is installed - This would display error logs in the CFO pod. Once the RayCluster CRD is installed, the CFO manager and RC Controller will start